sdk/core/azure-core-amqp/src/models/amqp_properties.cpp

```diff
@@ -119,14 +119,14 @@ namespace Azure { namespace Core { namespace Amqp { namespace Models {
     {
       if (properties_set_message_id(returnValue.get(), properties.MessageId.Value()))
       {
-        throw std::runtime_error("Could not set message id"); // LCOV_EXCL_LINE
+        throw std::runtime_error("Could not set message id");
```

Comment:
Coming up with a general strategy on how to test all invalid responses would be a good item for MQ

Reply:
That's what mocking is for :). Mocking allows you to trigger the underlying code to generate responses that are not normally possible.

And it should theoretically be possible to mock uAMQP since it's set up to be mockable. For our other components it will be significantly harder (we'd have to define mocks for OpenSSL, windows BCrypt, HTTP, etc).

OTOH, our current coverage on HTTP transports is rather low and I strongly suspect we can fix that.